frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -422,8 +512,72 @@ export const ClusterServiceVersionList: React.SFC<ClusterServiceVersionListProps
   );
   const NoDataEmptyMsg = () => <MsgBox title="No Operators Found" detail={noDataDetail} />;
 
-  return (
+  // remove duplicate global operator entries
```

Comment:
Instead of a comment, use a named helper function:

```javascript
const filteredData = removedDuplicateGlobalOperators(data);
```
Comments may be removed later, but it is unlikely someone will rename a function in a way that no longer describes its role.

Reply:
You may want to consider several helpers, to self-document this block:

```typescript
const filteredData = removedDuplicateGlobalOperators(data);

const removedDuplicateGlobalOperators = (operators) => {
  return operators.length > 0
          // filter really responds to true/false 
          ? operators.filter((source) => {
              const kind = referenceFor(source);
              if(isFoo(kind)) {
                return false 
              }
              if(isBar(kind)) {
                return true
              }
              if(isBaz(kind)) {
                return false;
              }
          })
          : [];
} 
```